src/modules/user/user.service.ts

```diff
@@ -41,4 +46,25 @@ export class UserService {
     theUser.password = await hash(password, salt);
     return plainToClass(UserResponseDto, await theUser.save());
   }
+
+  async disable(userId: number) {
+    const user = await this._userRepository.findOne({ id: userId });
+    if (user) throw new NotFoundException('user_not_found');
+    user.isActive = false;
+    return user.save();
+  }
+  async update(userId: number, newUser: UserUpdateDto) {
+    const user = await this._userRepository.findOne({ id: userId });
+    if (user) throw new NotFoundException('user_not_found');
+    if (user.id != userId) throw new ConflictException('user_not_match');
+    user.firstname = newUser.firstname;
+    user.lastname = newUser.lastname;
+    return plainToClass(UserResponseDto, await user.save());
+  }
+
+  async uploadAvatar(userId: number, avatarUrl: string) {
+    const user = await this._userRepository.findOne({ id: userId });
```

Comment:
What happen if the user is undefined?

Reply:
I used exceptions to handle undefined user but I've always got error when uploading an image even though the image is saved. Without the next exception, the api works fine, with them I always got error even when everything worked well
if (user) throw new NotFoundException('user_not_found');
if (user.id != userId) throw new ConflictException('user_not_match');